### PR TITLE
[ci/.bazelrc] Add color and reduce noisiness of Bazel output

### PR DESCRIPTION
**Pull request by dmcardle**
I'm curious what folks will think about this. I think these changes will improve the quality of the logs, but I recognize that it's more opinion that fact :)

(More details in commit messages.)

### CODE REVIEW

**Review by milesdai — APPROVED** [review]
Thanks!

**Comment by drewmacrae** [comment]
@mcy originally set color to no and I think it's for some subset of managing the log's size (it's probably not that one), keeping the logs generally compatible with other systems, processes and scripts, and keeping from interfering with azure's search functionality.

File: ci/.bazelrc [hunk 1]
```diff
@@ -4,12 +4,13 @@
 
 # These options disable the Bazel progress UI, resulting in output closer
 # to ninja and similar tools.
-common --color=no --curses=yes
+common --curses=no
```
Comment (drewmacrae) [inline ci/.bazelrc:7]: I cannot imagine why curses was being set to yes.

**Comment by dmcardle** [comment]
> @mcy originally set color to no and I think it's for some subset of managing the log's size (it's probably not that one), keeping the logs generally compatible with other systems, processes and scripts, and keeping from interfering with azure's search functionality.

* Size: I agree the color codes are unlikely to contribute much to the binary size. If it's an issue, there are a lot of other things we can do to offset the size, like silencing `updatemem`'s debug output to save ~100k lines. (I have more thoughts on `updatemem` output — I'll try to file an issue today.)

* Search: Azure Pipelines seems to have no problem searching the log. For example, I can search for "PASSED" and "FAILED", which show up [green and red](https://dev.azure.com/lowrisc/opentitan/_build/results?buildId=101775&view=logs&j=421a2038-410f-5c10-767f-4482292362ca&t=fe805732-5e37-5249-8636-c69eaa5ddf8e&l=93648).

* Compatibility: Only humans consume the CI logs, right? The only specific example I have of the ANSI color codes getting in the way is when I click "View raw log". Unsurprisingly, my browser does not interpret the color codes when rendering plain text, so it does muddy the content. On the other hand, it's trivial to do `curl "$LOG_URL | less -R` and see the colors in your terminal.

I have a slight preference for colors, but I'm happy to drop it from this PR if anyone has concerns.

**Comment by drewmacrae** [comment]
The way to miss things in search is to search across a color boundary, like for " PASSED", but it may not be important.

We also copy and paste from the logs, which can make the colors annoying.

But those are all nits that I simply wanted to discuss here before committing to this.

**Review by drewmacrae — APPROVED** [review]

**Pull request merged**